### PR TITLE
Debugger internal code cleanup

### PR DESCRIPTION
**Pull request by mlvdv**
No functional change; clean out remnants of earlier experiments and conversion to new Instrumentation.  Revise internal tracing to better support tuning of "expected behavior" during stepping.

### CODE REVIEW

File: truffle/com.oracle.truffle.api.debug/src/com/oracle/truffle/api/debug/Debugger.java [hunk 1]
```diff
@@ -684,68 +675,6 @@ protected void unsetStrategy() {
...
-    // unfinishedStepCount, stackDepth, currentStackDepth);
-    // // Should run in fast path
-    // if (unfinishedStepCount <= 0) {
-    // halt(node, mFrame, false);
-    // }
-    // strategyTrace("RESUME AFTER", "");
-    // }
-    // }
-    // }, "Debugger StepOver");
-
```
Comment (chumer) [inline truffle/com.oracle.truffle.api.debug/src/com/oracle/truffle/api/debug/Debugger.java]: Remove commented code.

Comment (mlvdv) [inline truffle/com.oracle.truffle.api.debug/src/com/oracle/truffle/api/debug/Debugger.java]: Thanks; missed that one.

**Comment by chumer** [comment]
Where are the unit tests explicitly testing this?
DebugInstrumentTest seems to be empty.

**Comment by mlvdv** [comment]
>Where are the unit tests explicitly testing this?

This is a code cleanup with no functional changes.

File: truffle/com.oracle.truffle.api.debug/src/com/oracle/truffle/api/debug/Debugger.java [hunk 2]
```diff
@@ -966,45 +872,55 @@ public FrameInstance visitFrame(FrameInstance frameInstance) {
...
-         * } }
-         */
-
-        void contextTrace(String format, Object... args) {
+        private void trace(String format, Object... args) {
             if (TRACE) {
-                final String srcName = (source != null) ? source.getName() : "no source";
-                Debugger.trace("<%d> %s (%s)", level, String.format(format, args), srcName);
+                String location = "";
+                if (haltedEventContext != null && haltedEventContext.getInstrumentedNode().getSourceSection() != null) {
```
Comment (chumer) [inline truffle/com.oracle.truffle.api.debug/src/com/oracle/truffle/api/debug/Debugger.java]: Please use `haltedEventContext.getInstrumentedSourceSection()` instead.

**Comment by chumer** [comment]
@mlvdv I completely understand that this is out of scope of this PR. But how would I verify that your changes don't break anything?

BTW.: I still see commented code which prevents me from applying the accept label.

**Comment by mlvdv** [comment]
@chumer  I don't see any remaining comment out code; one of the goals of this cleanup was to remove all of that.  Can you point me to specific locations in the three modified files?

**Comment by chumer** [comment]
@mlvdv just click on the "files changed" tab you see it there. probably your debugger-cleanup branch is not up-to-date for mlvdv/truffle?

**Comment by mlvdv** [comment]
@chumer I actually have clicked on that tab and all I see are the places where I removed such things.

**Comment by jtulach** [comment]
Martin @entlicher, I guess you are OK with @mlvdv's changes, right?

**Comment by entlicher** [comment]
@jtulach yes, I'm fine with these changes.

**Pull request merged**